src/components/course/tests/CourseSkills.test.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { AppContext } from '@edx/frontend-platform/react';
+import { screen, render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import { CourseContextProvider } from '../CourseContextProvider';
+import CourseSkills from '../CourseSkills';
+
+import { SKILLS_BUTTON_TEXT } from '../data/constants';
+
+/* eslint-disable react/prop-types */
+const CourseSkillsWithContext = ({
+  initialAppState = {},
+  initialCourseState = {},
+}) => (
+  <AppContext.Provider value={initialAppState}>
+    <CourseContextProvider initialState={initialCourseState}>
+      <CourseSkills />
+    </CourseContextProvider>
+  </AppContext.Provider>
+);
+/* eslint-enable react/prop-types */
+
+describe('<CourseSkills />', () => {
+  const initialAppState = {
+    enterpriseConfig: {
+      slug: 'test-enterprise-slug',
+    },
+  };
+  const initialCourseState = {
+    course: {
+      skillNames: ['test-skill1', 'test-skill2', 'test-skill3', 'test-skill4'],
+    },
+  };
+
+  test('renders course skills less than limit', () => {
+    render(
+      <CourseSkillsWithContext
+        initialAppState={initialAppState}
+        initialCourseState={initialCourseState}
+      />,
+    );
+    initialCourseState.course.skillNames.forEach((skill) => {
+      expect(screen.queryByText(skill)).toBeVisible();
+    });
+  });
+
+  test('renders course skills greater than limit', () => {
+    const skillGreaterThanLimit = ['test-skill1', 'test-skill2', 'test-skill3', 'test-skill4', 'test-skill5', 'test-skill6'];
+    const courseState = {
+      ...initialCourseState,
+      course: {
+        ...initialCourseState.course,
+        skillNames: skillGreaterThanLimit,
+      },
+    };
+    render(
+      <CourseSkillsWithContext
+        initialAppState={initialAppState}
+        initialCourseState={courseState}
+      />,
+    );
+    initialCourseState.course.skillNames.forEach((skill, index) => {
+      if (index >= 5) {
```

Comment:
SUGGEST: Usually conditions in tests are considered smell: https://testsmells.org/pages/testsmells.html#ConditionalTestLogic, I still think it is better to have either two separate test cases for these to replace tests's if conditions, 

OR,

(slightly less preferred but still fine) if you would rather these be one test, use a slight change to try is sth like two assertions

```
const showSkills = ...filter.. to get... these ..
const hiddenSkills = ...filter.. to get .. these ..
```

then it can also become also a bit less conditional-ly...

```
expect(...) for shownSkills

expect(...) for hiddenSkills
```

Reply:
But even if I split the test cases, I will still have to check this condition. I'm not sure how we can achieve a condition-free test case in this scenario. The only solution that comes to my mind is slicing the arrays.

Reply:
actually now that I think if it your test case here is to check that N of the M cases are hidden others are shown.. so now I almost think second option is better since you are testing one thing .. (and I noticed you already have a test case for shown skills above so no point adding another case)
```
const shownSkills = initialCourseState.course.skillNames.slice(0,5)
const hiddenSkills = initialCourseState.course.skillNames.slice(5,6)
expect(...) for shownSkills
expect(...) for hiddenSkills
```

Sorry was not clear on filter part, instead of filter, it would be a slice like this. My example may not be accurate but you get the idea